### PR TITLE
Replace infinispan with the lightweight caffeine

### PR DESCRIPTION
**Pull request by sswguo**
We plan to integrate this as a sidecar to the mw pipeline, and have to slim the image to avoid the performance issue when download the image each time for mw pipeline running.  We can do this from app level and image level, this PR is for app level, I found two big deps, one is quarkus-opentelemetry and another one is infinispan.

### CODE REVIEW

**Review by ruhan1 — COMMENTED** [review]
LGTM

File: src/main/java/org/commonjava/indy/service/httprox/util/CacheProducer.java [hunk 1]
```diff
@@ -15,44 +15,38 @@
...
+        return caches.computeIfAbsent( named, ( k ) -> buildCache());
     }
 
-    private Cache buildCache(String named)
+    private Cache buildCache()
     {
-        cacheManager.defineConfiguration( named, new ConfigurationBuilder().build() );
-        return cacheManager.getCache(named, Boolean.TRUE);
+        Cache<String, String> cache = Caffeine.newBuilder()
+                    .expireAfterAccess(15, TimeUnit.MINUTES)
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/httprox/util/CacheProducer.java:47]: Should we set a maximumsize for the cache to vaoid potential OOM issue?

Comment (sswguo) [reply src/main/java/org/commonjava/indy/service/httprox/util/CacheProducer.java:47]: Yeah, good point, just in case.

**Review by ligangty — COMMENTED** [review]
LGTM

**Comment by ligangty** [comment]
BTW, seems caffeine has quarkus extentions here: https://quarkus.io/extensions/io.quarkus/quarkus-caffeine/?tab=docs

**Comment by sswguo** [comment]
> https://quarkus.io/extensions/io.quarkus/quarkus-caffeine/?tab=docs

Cool, seems it's quite clean as the follows. I will check again and ref it. 
`
[INFO] +- io.quarkus:quarkus-caffeine:jar:3.6.9:compile
[INFO] |  \- com.github.ben-manes.caffeine:caffeine:jar:3.1.5:compile
`

**Pull request merged**